### ISSUE TITLE
Code getting struck when extracting Mask-RCNN Features

### ISSUE THREAD

**Issue by GuptaSonam** [facebookresearch/instpred#6]
Hi, 
Thanks for sharing the code with detailed instructions about the setup. 
I am trying to train the model from scratch but it gets stuck at when computing Mask R-CNN features with the last lines in the log file as :
INFO - - Cityscapes dataset, size : 500
{'feat_type': 'fpn_res5_2_sum', 'it': 2975, 'n_target_frames': 1, 'frame_ss': 3, 'nb_features': 256, 'n_input_frames': 4}
{'n_target_frames': 1, 'frame_ss': 3, 'nb_features': 256, 'feat_type': 'fpn_res5_2_sum', 'n_input_frames': 4, 'it': 500}
[I net_async_base.h:205] Using specified CPU pool size: 8; device id: -1
[I net_async_base.h:210] Created new CPU pool, size: 8; device id: -1

I am using the system with 2 GPU. Not sure why is this taking CPUs for extracting feature using the pre-trained Mask R-CNN  model that you have provided in the repo. nvidia-smi shows no GPU utilization and the process is stuck here even after 12 hrs. 

I would appreciate any inputs that you may have so that I can fix this issue.

Thanks

**Comment by guptasonam1602** [comment]
I reinstalled caffe 2 and it works now. However it uses only a single gpu although I have 4  gpus available.

**Comment by birdflyto** [comment]
> Hi,
> Thanks for sharing the code with detailed instructions about the setup.
> I am trying to train the model from scratch but it gets stuck at when computing Mask R-CNN features with the last lines in the log file as :
> INFO - - Cityscapes dataset, size : 500
> {'feat_type': 'fpn_res5_2_sum', 'it': 2975, 'n_target_frames': 1, 'frame_ss': 3, 'nb_features': 256, 'n_input_frames': 4}
> {'n_target_frames': 1, 'frame_ss': 3, 'nb_features': 256, 'feat_type': 'fpn_res5_2_sum', 'n_input_frames': 4, 'it': 500}
> [I net_async_base.h:205] Using specified CPU pool size: 8; device id: -1
> [I net_async_base.h:210] Created new CPU pool, size: 8; device id: -1
> 
> I am using the system with 2 GPU. Not sure why is this taking CPUs for extracting feature using the pre-trained Mask R-CNN model that you have provided in the repo. nvidia-smi shows no GPU utilization and the process is stuck here even after 12 hrs.
> 
> I would appreciate any inputs that you may have so that I can fix this issue.
> 
> Thanks

Hi @guptasonam1602 ，i have been tried to run the model for half of month,but i still stuck in the getting precomputed features by the given mask-rcnn model. The envs is ok for all the train and test on on the sub network of Detecron . could you share some detail how to run precompute_maskrcnn_backbone_features.py (maybe there is some option ignored in the repo) to get the precomputed features. And where to insert the pretrained instance segmentation model?
I will appreciiate it if you can give some advices and help.
